avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -960,7 +979,8 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
       // and name().
       if (schema.name() != null) {
         if (Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())
-            && schema.parameters().containsKey(CONNECT_AVRO_DECIMAL_PRECISION_PROP)) {
+            && (schema.parameters().containsKey(CONNECT_AVRO_DECIMAL_PRECISION_PROP)
+                || forceLegacyDecimal)) {
```

Comment:
If we added `CONNECT_AVRO_DECIMAL_PRECISION_PROP` to the schema parameters in the block above, which only happens when `Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())`, then we wouldn't need to modify this block.

Reply:
Yeah, originally tried that. Failed during testing because `schema.paramaters()` returned an immutable map.